skrf/network.py

```diff
@@ -5461,20 +5484,24 @@ def innerconnect(ntwkA: Network, k: int, l: int, num: int = 1) -> Network:
     if (l + num - 1 > ntwkA.nports - 1):
         raise IndexError('Port `l` out of range')
 
+    # 'power' is not supported, convert to supported definition and back afterwards
+    if ntwkA.s_def == 'power':
+        ntwkA = ntwkA.copy()
+        ntwkA.renormalize(ntwkA.z0, 'pseudo')
```

Comment:
I don't understand

Reply:
Previously, we were duplicating the entire `ntwkA` as `ntwkC` upfront and then pre-processing `ntwkC` for **definition** and **port impedance**, this will led to a significant cost in terms of s-parameter overwriting for `ntwkC`.

https://github.com/scikit-rf/scikit-rf/blob/3e92b2cd73600919ed6ce81dbe36ec61e98d087b/skrf/network.py#L5464-L5483

With the implementation of `shallow_copy=True`, `ntwkC` doesn't holds the complete s-parameters data; instead, `ntwkA` retains this information. As a result, the previous checking for **definition** and **port impedance** should be  conducted at `ntwkA` instead.

Therefore
1. We're sticking with the variable name `ntwkA` to keep the program running and avoid deepcopy of `s-parameters ndarray` as much as possible. If definition and port impedance checking are passed, it will step from ```ntwkC = ntwkA.copy(shallow_copy=True)``` directly to ```ntwkC.s = innerconnect_s(ntwkA.s, k, l)```, thereby avoiding `ntwkC.s` from being overwritten to memory during this process.

2. If definition checking is executed, since `renormalize()` is in-place, we copy a new object for renormalization but still name it `ntwkA`, so as to ensure that the correct `s-parameters` can be used in the subsequent impedance check and `innerconnect_s()` operations on `ntwkA`.

In summary, creating a copy with ```ntwkA = ntwkA.copy()``` and naming it `ntwkA` is essential because, as we've discussed, `ntwkA` needs to carry the s-parameters data. All operations before `innerconnect_s()` must be performed on the variable named `ntwkA`, and using `copy()` to prevent the original data from being corrupted.